src/main.rs

```diff
@@ -1,42 +1,137 @@
-use std::fs;
-use clap::{Args, Parser};
+use clap::Parser;
 use parcel_css::stylesheet::{MinifyOptions, ParserOptions, PrinterOptions, StyleSheet};
+use std::{ffi, fs, io, path};
+use serde::Serialize;
 
 #[derive(Parser, Debug)]
-#[clap(author, version, about, long_about = None)]
-enum Cli {
-  /// Minify a given CSS file
-  Minify(MinifyArgs),
+#[clap(author, about, long_about = None)]
+struct CliArgs {
+    /// Target CSS file
+    input_file: String,
+    /// Destination file for the output
+    #[clap(short, long, group = "output_file")]
+    output_file: Option<String>,
+    /// Minify the output
+    #[clap(short, long)]
+    minify: bool,
+    /// Enable parsing CSS nesting
+    #[clap(short, long)]
+    nesting: bool,
+    /// Enable CSS modules in output
+    #[clap(short, long, group = "css_modules")]
+    css_modules: bool,
+    /// Default: <output_file>.json
+    #[clap(long, requires = "css_modules")]
+    css_modules_output_file: Option<String>,
+    /// Enable sourcemap, at <output_file>.map
+    #[clap(short, long, requires = "output_file")]
+    sourcemap: bool,
 }
 
-#[derive(Args, Debug)]
-struct MinifyArgs {
-  /// The CSS file to minify
-  input_file: String,
-  /// The name of the output file
-  #[clap(short, long)]
-  output_file: Option<String>,
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct SourceMapJson<'a> {
+  version: u8,
+  mappings: String,
+  sources: &'a Vec<String>,
+  sources_content: &'a Vec<String>,
+  names: &'a Vec<String>
 }
 
 pub fn main() -> Result<(), std::io::Error> {
-  match Cli::parse() {
-    Cli::Minify(cli_args) => {
-      minify_css(cli_args)
+    let cli_args = CliArgs::parse();
+    let source = fs::read_to_string(&cli_args.input_file)?;
+
+    let filename = path::Path::new(&cli_args.input_file)
+        .file_name()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .into();
```

Comment:
I think we should probably avoid this. For example, if you had two different files in your project each named `index.css` or something but in different folders, the class names could clash. Using the full absolute path (or a relative one from some project root) avoids this.

Reply:
Oh, that's a good point, that does seem like it will definitely happen, especially once there's multiple file support!

I guess a pertinent question is whether invoking from different locations, the output should be deterministic. I still think yes (though, it's clear that `file_name` doesn't do the job here). Is there something else to hash that makes sense? The source file contents?